core/sdl/sdl.cpp

```diff
@@ -158,18 +158,26 @@ void input_sdl_init()
 			SDL_SetHint(SDL_HINT_XINPUT_ENABLED, "0");
 		}
 #endif
-		if (SDL_InitSubSystem(SDL_INIT_JOYSTICK) < 0)
-			die("SDL: error initializing Joystick subsystem");
-
 		std::string db = get_readonly_data_path("gamecontrollerdb.txt");
 		int rv = SDL_GameControllerAddMappingsFromFile(db.c_str());
+//#ifdef _ENABLE_EMUELEC
+		if (rv < 0)
+		{
+			db = (std::string) nowide::getenv("SDL_GAMECONTROLLERCONFIG_FILE");
```

Comment:
`getenv` can return `nullptr` and assigning it to a `std::string` is undefined behavior.